### PR TITLE
Fixing deprecations in previously not included functions

### PR DESCRIPTION
**Pull request by Brinky5678**
Dear Maintainers, 

Within the "Spice.jl" module definition, I included the files "constants.jl", "math.jl", "pck.jl", and "spk.jl". I required the use of the functions already created in these files, yet were previously not exported. Additionally, some deprecation warnings were fixed in those files regarding array definitions.  

Please consider my changes. 

With kind regards,

Dennis Brinkman

### CODE REVIEW

**Comment by helgee** [comment]
Thank you for your contribution! Could I bother you to move the functions you need to the respective alphabetical file? I orginally wanted to group them by functionality but lost track due to the large number of functions in SPICE.

If you don't have the time I will merge and move them myself.

**Comment by Brinky5678** [comment]
No Problem. Somewhere this week I will move everything to be in the correct alphabetical order. Additionally, I will also include some tests of the functions to improve the code coverage.

**Comment by Brinky5678** [comment]
Updated the branch, including additional tests. Not sure about the codecov/patch, maybe I don't fully understand how that features works yet.

File: test/m.jl [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#= mxvg is Severely Broken, test below does not work and dont know why!
+ Perform the following calculation: 1*1 + 2*2 + 1*3, 3*1 + 1*2 + 4*3
+@testset "M" begin 
+    matrix = [1. 1. 1. 
+                2. 3. 4.]
+    vecgood = [1., 2., 3.]
+    @test mxvg(matrix, vecgood) == [6.,20.]
```
Comment (helgee) [inline test/m.jl:5]: The problem is that CSPICE expects the matrix to be in row-major order whereas Julia uses column-major order.
We need to pass the transpose of the input matrix to the C code, e.g.:
```julia
function mxvg(m1, v2)
    lm1, lm2 = size(m1)
    lv = length(v2)
    if lm2 != lv
        error("Dimension mismatch.")
    end
    vout = Array{Float64}(lm1)
    ccall((:mxvg_c, libcspice), Void, (Ptr{Float64}, Ptr{Float64}, Cint, Cint, Ptr{Float64}),
    #= use transpose =# m1', v2, lm1, lm2, vout)
    handleerror()
    return vout
end
```

**Comment by Brinky5678** [comment]
That was indeed the problem! It is working now.

**Comment by helgee** [comment]
Please activate the test and then this is good to go 👍 Thank you!

(BTW: You might want to add the email address you use for your Git commits to your GitHub account. This way they become associated to your account and become clickable in the interface.)

**Pull request merged**